src/components/MatchRuleList.tsx

```diff
@@ -0,0 +1,255 @@
+import { useTranslation } from 'react-i18next'
+import { Translation, Budget, Theme } from '../types'
+import { useEffect, useState } from 'react'
+import {
+  ArrowLongDownIcon,
+  ArrowLongRightIcon,
+  CloudArrowUpIcon,
+  PlusIcon,
+  QuestionMarkCircleIcon,
+  TrashIcon,
+} from '@heroicons/react/24/outline'
+import { Bars2Icon } from '@heroicons/react/24/solid'
+import Error from './Error'
+import FormField from './FormField'
+import { api } from '../lib/api/base'
+import { MatchRule, Account } from '../types'
+import { ReactSortable } from 'react-sortablejs'
+import Autocomplete from './Autocomplete'
+import { safeName } from '../lib/name-helper'
+import LoadingSpinner from './LoadingSpinner'
+
+const matchRuleApi = api('matchRules')
+const accountApi = api('accounts')
+
+type Props = {
+  budget: Budget
+  setNotification: (notification: string) => void
+}
+
+const MatchRuleList = ({ budget, setNotification }: Props) => {
+  const { t }: Translation = useTranslation()
+
+  const [error, setError] = useState('')
+  const [isLoading, setIsLoading] = useState(true)
+  const [hideHelp, setHideHelp] = useState(true)
+
+  const [matchRules, setMatchRules] = useState<MatchRule[]>([])
+  const [accounts, setAccounts] = useState<Account[]>([])
+  const accountGroups = [
+    {
+      title: t('accounts.internalAccounts'),
+      items: accounts.filter(account => !account.external),
+    },
+    {
+      title: t('accounts.externalAccounts'),
+      items: accounts.filter(account => account.external),
+    },
+  ]
+
+  const addMatchRule = () => {
+    // The ID we set here is needed for uniqueness in the sortable list, but ignored later on when we actually send the data to the backend.
+    // The priority is in order from top to bottom and can be 0, so we can just use the length of the matchRules.
+    //
+    // We also set the links to a dummy value since we only need them for match rules loaded from the backend
+    setMatchRules([
+      { id: matchRules.length.toString(), links: {} },
+      ...matchRules,
+    ])
+  }
+
+  const deleteMatchRule = (id: string) => {
+    console.log(id)
+    let newMatchRules = [...matchRules]
+    const ruleIndex = matchRules.findIndex(rule => rule.id === id)
+
+    newMatchRules[ruleIndex].toDelete = true
+    console.log(newMatchRules)
+    setMatchRules(newMatchRules)
+  }
+
+  const saveMatchRules = async () => {
+    let updatedMatchRules: MatchRule[] = []
```

Comment:
You only push into the array, and never re-assign the variable.
```suggestion
    const updatedMatchRules: MatchRule[] = []
```

Reply:
Fixed with 9d33505